vehicle/steer_offset_estimator/src/steer_offset_estimator_node.cpp

```diff
@@ -0,0 +1,136 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "steer_offset_estimator/steer_offset_estimator_node.hpp"
+
+#include "vehicle_info_util/vehicle_info_util.hpp"
+
+#include <memory>
+#include <utility>
+
+namespace steer_offset_estimator
+{
+SteerOffsetEstimatorNode::SteerOffsetEstimatorNode(const rclcpp::NodeOptions & node_options)
+: Node("steer_offset_estimator", node_options)
+{
+  using std::placeholders::_1;
+  // get parameter
+  const auto vehicle_info = vehicle_info_util::VehicleInfoUtil(*this).getVehicleInfo();
+  wheel_base_ = vehicle_info.wheel_base_m;
+  covariance_ = this->declare_parameter("initial_covariance", 1000.0);
+  forgetting_factor_ = this->declare_parameter("forgetting_factor", 0.999);
+  update_hz_ = this->declare_parameter<double>("steer_update_hz", 10.0);
+  valid_min_velocity_ = this->declare_parameter<double>("valid_min_velocity");
+  valid_max_steer_ = this->declare_parameter<double>("valid_max_steer");
+  warn_steer_offset_ = this->declare_parameter<double>("warn_steer_offset_deg") * M_PI / 180.0;
+
+  // publisher
+  pub_steer_offset_ = this->create_publisher<Float32Stamped>("~/output/steering_offset", 1);
+  pub_steer_offset_cov_ =
+    this->create_publisher<Float32Stamped>("~/output/steering_offset_covariance", 1);
+
+  // subscriber
+  sub_twist_ = this->create_subscription<TwistStamped>(
+    "~/input/twist", 1, std::bind(&SteerOffsetEstimatorNode::onTwist, this, _1));
+  sub_steer_ = this->create_subscription<Steering>(
+    "~/input/steer", 1, std::bind(&SteerOffsetEstimatorNode::onSteer, this, _1));
+
+  /* Diagnostic Updater */
+  updater_ptr_ = std::make_shared<Updater>(this, 1.0 / update_hz_);
+  updater_ptr_->setHardwareID("steer_offset_estimator");
+  updater_ptr_->add("steer_offset_estimator", this, &SteerOffsetEstimatorNode::monitorSteerOffset);
+
+  // timer
+  {
+    auto on_timer = std::bind(&SteerOffsetEstimatorNode::onTimer, this);
+    const auto period = rclcpp::Rate(update_hz_).period();
+    timer_ = std::make_shared<rclcpp::GenericTimer<decltype(on_timer)>>(
+      this->get_clock(), period, std::move(on_timer),
+      this->get_node_base_interface()->get_context());
+    this->get_node_timers_interface()->add_timer(timer_, nullptr);
+  }
+}
+
+// function for diagnostics
+void SteerOffsetEstimatorNode::monitorSteerOffset(DiagnosticStatusWrapper & stat)
+{
+  using DiagStatus = diagnostic_msgs::msg::DiagnosticStatus;
+  const double eps = 1e-3;
+  if (covariance_ > eps) {
+    stat.summary(DiagStatus::OK, "Preparation");
+    return;
+  }
+  if (estimated_steer_offset_ > warn_steer_offset_) {
```

Comment:
```suggestion
  if (std::abs(estimated_steer_offset_) > warn_steer_offset_) {
```
I noticed that the word `warn_steer_offset_` is ambiguous in that it is absolute value or not.
This suggestion is on the premise of absolute value.